code/modules/telesci/telepad.dm

```diff
@@ -116,59 +145,85 @@
 
 ///HANDHELD TELEPAD USER///
 /obj/item/weapon/rcs
-	name = "rapid-crate-sender (RCS)"
-	desc = "Use this to send crates and closets to cargo telepads."
+	name = "cargo transporter"
+	desc = "A device for teleporting crated goods."
 	icon = 'icons/obj/telescience.dmi'
 	icon_state = "rcs"
 	flags = CONDUCT
 	force = 10
 	throwforce = 10
 	throw_speed = 2
 	throw_range = 5
-	var/rcharges = 10
-	var/obj/machinery/pad = null
-	var/last_charge = 30
-	var/mode = 0
-	var/rand_x = 0
-	var/rand_y = 0
-	var/emagged = 0
-	var/teleporting = 0
-
-/obj/item/weapon/rcs/New()
-	..()
-	START_PROCESSING(SSobj, src)
+
+	var/charges = 10
+	var/maximum_charges = 10.0
+	var/obj/machinery/telepad_cargo/target_pad = null
+	var/target_turf = null
 
 /obj/item/weapon/rcs/examine(mob/user)
 	..()
-	user << "There are [rcharges] charge\s left."
+	user << "There are [charges]/[maximum_charges] charge\s left."
 
-/obj/item/weapon/rcs/Destroy()
-	STOP_PROCESSING(SSobj, src)
-	return ..()
-/obj/item/weapon/rcs/process()
-	if(rcharges > 10)
-		rcharges = 10
-	if(last_charge == 0)
-		rcharges++
-		last_charge = 30
+/obj/item/weapon/rcs/attack_self(mob/user)
+	if (charges < 1)
+		user << "<span style=\"color:red\">The transporter is out of charge.</span>"
+		return
+	if (!cargopads.len) usr << "<span style=\"color:red\">No receivers available.</span>"
 	else
-		last_charge--
+	//here i set up an empty var that can take any object, and tell it to look for absolutely anything in the list
+		var/selection = input("Select Cargo Pad Location:", "Cargo Pads", null, null) as null|anything in cargopads
+		if(!istype(selection, /obj/machinery/telepad_cargo))
+			return
+		
+		var/obj/machinery/telepad_cargo/P = selection
+		
+		if(!selection)
+			return
+		var/turf/T = get_turf(selection)
+		//get the turf of the pad itself
+		if (!T)
+			usr << "<span style=\"color:red\">Target not set!</span>"
+			return
+		usr << "Target set to [T.loc]."
+		//blammo! works!
+		target_pad = P
+		target_turf = T
+
+/obj/item/weapon/rcs/proc/cargoteleport(var/obj/T, var/mob/user)
+	if (!target_turf)
+		user << "<span style=\"color:red\">You need to set a target first!</span>"
+		return
+	if (!target_pad.active)
+		user << "<span style=\"color:red\">Unable to connect to remote pad!</span>"
+	if (charges < 1)
+		user << " <span style=\"color:red\">The transporter is out of charge.</span>"
+		return
 
-/obj/item/weapon/rcs/attack_self(mob/user)
-	if(emagged)
-		if(mode == 0)
-			mode = 1
-			playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
-			user << "<span class='caution'>The telepad locator has become uncalibrated.</span>"
+	user << "<span style=\"color:blue\">Teleporting [T]...</span>"
+	playsound(user.loc, "sound/machines/click.ogg", 50, 1)
+
+	target_pad.icon_state = "pad-beam"
+
+	if(do_after(user, 50, target=T, progress = 1))

```

Comment:
`do_after` can be stacked and you only remove charges after `do_after` has finished, allowing the user to use the item more often